lib/td/client/api/job.rb

```diff
@@ -201,29 +197,29 @@ def job_result_each(job_id, &block)
   # @param [Proc] block
   # @return [nil]
   def job_result_each_with_compr_size(job_id, &block)
-    get("/v3/job/result/#{e job_id}", {'format'=>'msgpack'}) {|res|
-      if res.code != "200"
+    upkr = MessagePack::Unpacker.new
+    infl = nil
+
+    get("/v3/job/result/#{e job_id}", {'format'=>'msgpack'}) {|res, chunk, current_total_chunk_size|
+      if res.code != 200
         raise_error("Get job result failed", res)
       end
 
-      res.extend(DirectReadBodyMixin)
-      if res.header['Content-Encoding'] == 'gzip'
+      # default to decompressing the response since format is fixed to 'msgpack'
+      if !infl && (res.header['Content-Encoding'].include? 'gzip')
```

Comment:
yes. This is what I will be rewritten as follows.

```
infl ||= Zlib::Infrate.new
```

now, I fixed it.

https://github.com/treasure-data/td-client-ruby/commit/aedd61c66472c256b4498a0b64059c13dbbe77da